pkg/provider/akeyless/akeyless.go

```diff
@@ -404,6 +409,118 @@ func (a *Akeyless) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecre
 	return secretData, nil
 }
 
+func (a *Akeyless) SecretExists(ctx context.Context, ref esv1beta1.PushSecretRemoteRef) (bool, error) {
+	if utils.IsNil(a.Client) {
+		return false, errors.New(errUninitalizedAkeylessProvider)
+	}
+	secret, err := a.GetSecret(ctx, esv1beta1.ExternalSecretDataRemoteRef{Key: ref.GetRemoteKey()})
+	if errors.Is(err, ErrItemNotExists) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	if ref.GetProperty() == "" {
+		return true, nil
+	}
+	var secretMap map[string]any
+	err = json.Unmarshal(secret, &secretMap)
+	if err != nil {
+		return false, err
+	}
+	_, ok := secretMap[ref.GetProperty()]
+	return ok, nil
+}
+
+func (a *Akeyless) PushSecret(ctx context.Context, secret *corev1.Secret, psd esv1beta1.PushSecretData) error {
+	if utils.IsNil(a.Client) {
+		return errors.New(errUninitalizedAkeylessProvider)
+	}
+	ctx, err := a.contextWithToken(ctx)
+	if err != nil {
+		return err
+	}
+	secretRemote, err := a.GetSecret(ctx, esv1beta1.ExternalSecretDataRemoteRef{Key: psd.GetRemoteKey()})
+	isNotExists := errors.Is(err, ErrItemNotExists)
+	if err != nil && !isNotExists {
+		return err
+	}
+	var data map[string]any
+	if isNotExists {
+		err = nil
+		mapSize := 1
+		if psd.GetProperty() == "" {
+			mapSize = len(secret.Data)
+		}
+		data = make(map[string]any, mapSize)
+	} else {
+		err = json.Unmarshal(secretRemote, &data)
+	}
+	if err != nil {
+		return err
+	}
+	if psd.GetProperty() == "" {
+		data = make(map[string]any, len(secret.Data))
+		for k, v := range secret.Data {
+			data[k] = string(v)
+		}
+	} else if v, ok := secret.Data[psd.GetSecretKey()]; ok {
+		data[psd.GetProperty()] = string(v)
+	}
```

Comment:
This section is super confusion, please rewrite it. Also you check psd.GetProperty twice and assign data twice. If the secret exists unmarshal, otherwise just move on. There is no point in doing that assign thing twice.

Also, you can use https://pkg.go.dev/maps package to collect the keys from secret.Data into data with one function call. Which is nice.

Reply:
Checking it out thanks

Reply:
I cant use Copy or All\Collect out of the box since the types are different so I need to do a for loop mapping in any case
I refactored the double assignment

Reply:
Thanks